### PR TITLE
V6 update with master

### PR DESCRIPTION
**Pull request by colinlohner**
**Overview**
Merge in latest master

**Screenshots (if applicable)**
n/a

**Documentation**
- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

File: src/menu/src/MenuItem.js [hunk 1]
```diff
@@ -55,6 +56,29 @@ const MenuItem = memo(
...
+      iconColor = 'disabled'
+    }
+
+    const textColor = intent
+
+    const secondaryTextColor = disabled ? textColor : 'muted'
+
+    const disabledProps = useMemo(() => {
+      return disabled ? {
+        backgroundColor: theme.colors.background.tint1,
```
Comment (akleiner2) [inline src/menu/src/MenuItem.js:71]: Wonder if it makes sense now to move this into the theme?

**Review by akleiner2 — APPROVED** [review]
One comment inline about moving something into the theme, but looks nice! 👍

File: src/autocomplete/src/Autocomplete.js [hunk 2]
```diff
@@ -53,6 +53,7 @@ const AutocompleteItems = ({
   title,
   width
 }) => {
+  itemsFilter = itemsFilter || fuzzyFilter(itemToString)
```
Comment (mshwery) [inline src/autocomplete/src/Autocomplete.js:56]: `itemsFilter` above shouldn't need the default assignment with this.

File: src/combobox/src/Combobox.js [hunk 3]
```diff
@@ -30,12 +30,19 @@ const Combobox = memo(function Combobox(props) {
...
       }
     }
+
+    if (
+      autocompleteProps &&
+      typeof autocompleteProps.onStateChange === 'function'
+    ) {
+      autocompleteProps.onStateChange(changes, stateAndHelpers)
+    }
   }, [])
```
Comment (mshwery) [inline src/combobox/src/Combobox.js:46]: You'll need to include `autocompleteProps` in the dep array I think.

File: src/menu/src/MenuItem.js [hunk 4]
```diff
@@ -67,21 +90,22 @@ const MenuItem = memo(
...
           icon={icon}
-          color={intent === 'none' ? 'default' : intent}
+          color={iconColor}
           marginLeft={16}
           marginRight={-4}
           size={16}
           flexShrink={0}
         />
-        <Text color={intent} marginLeft={16} marginRight={16} flex={1}>
+        <Text color={textColor} marginLeft={16} marginRight={16} flex={1}>
```
Comment (mshwery) [inline src/menu/src/MenuItem.js:104]: This change seems unnecessary.

File: src/menu/src/MenuItem.js [hunk 5]
```diff
@@ -67,21 +90,22 @@ const MenuItem = memo(
...
           {children}
         </Text>
         {secondaryText && (
-          <Text marginRight={16} color="muted">
+          <Text marginRight={16} color={secondaryTextColor}>
```
Comment (mshwery) [inline src/menu/src/MenuItem.js:108]: Unsure why this was changed.

File: src/menu/src/MenuItem.js [hunk 6]
```diff
@@ -55,6 +56,28 @@ const MenuItem = memo(
...
+
+    const disabledProps = useMemo(() => {
+      return disabled ? {
+        cursor: 'not-allowed',
+        disabled: true,
+        onClick: null,
+        onKeyPress: null,
+        tabIndex: -1,
+        'aria-disabled': 'true',
+        'data-isselectable': 'false'
```
Comment (mshwery) [inline src/menu/src/MenuItem.js:77]: Prefer this to happen below instead of through props spreading.

File: src/menu/src/MenuItem.js [hunk 7]
```diff
@@ -55,6 +56,28 @@ const MenuItem = memo(
...
+    const textColor = intent
+
+    const secondaryTextColor = disabled ? textColor : 'muted'
+
+    const disabledProps = useMemo(() => {
+      return disabled ? {
+        cursor: 'not-allowed',
+        disabled: true,
+        onClick: null,
+        onKeyPress: null,
```
Comment (mshwery) [inline src/menu/src/MenuItem.js:74]: These should probably be `undefined`? or maybe handled in the definitions of `handleClick` and `handleKeyPress`? (I realize these changes aren't from your PR, but revisiting them now).

File: src/popover/src/Popover.js [hunk 8]
```diff
@@ -60,11 +62,17 @@ const Popover = memo(
      * Methods borrowed from BlueprintJS
      * https://github.com/palantir/blueprint/blob/release/2.0.0/packages/core/src/components/overlay/overlay.tsx
      */
-    const bringFocusInside = () => {
+    const bringFocusInside = useCallback((e) => {
+      if(isShown && e) {
```
Comment (mshwery) [inline src/popover/src/Popover.js:66]: Good! Making sure this has the `&& e` patch.

File: src/tag-input/src/TagInput.js [hunk 9]
```diff
@@ -48,7 +48,7 @@ const TagInput = memo(
       onFocus,
       onInputChange,
       className,
-      inputProps,
+      inputProps = {},
```
Comment (mshwery) [inline src/tag-input/src/TagInput.js:51]: This should use `emptyObject` defined outside the function instead

File: src/textarea/src/TextareaField.js [hunk 10]
```diff
@@ -1,69 +1,71 @@
...
-        appearance={appearance}
-        placeholder={placeholder}
-        spellCheck={spellCheck}
-        {...remainingProps}
-      />
-    </FormField>
-  )
-})
+const TextareaField = memo(
+  forwardRef(function TextareaField(props, ref) {
```
Comment (mshwery) [inline src/textarea/src/TextareaField.js:9]: Looks like we are missing the `ref` usage?

**Review by mshwery — COMMENTED** [review]
Looks good! Left a few minor comments (most can be punted) but a few worth visiting.

File: src/textarea/src/TextareaField.js (hunk 10 above)
Comment (colinlohner) [reply src/textarea/src/TextareaField.js:9]: Fixed!

File: src/popover/src/Popover.js (hunk 8 above)
Comment (colinlohner) [reply src/popover/src/Popover.js:66]: 👍

Comment (akleiner2) [reply src/popover/src/Popover.js:66]: The formatting for this still feels weird (should be space after `if`), but nbd

**Review by akleiner2 — APPROVED** [review]
Cool - original things that needed to be addressed look good!

**Pull request merged**